In [26]:
import pandas as pd
import numpy as np

In [27]:
M = pd.read_csv("MovieReviews.csv", index_col='Name')

In [28]:
M.head()

American Sniper  The Hunger Games: Mockingjay - Part 1  \
Name                                                                         
Aarti Jaiswal                   4.0                                    NaN   
Aditya Dharmasagar              5.0                                    NaN   
Adrian Cavallaris               NaN                                    4.0   
Ahmed Muheebuddin               5.0                                    NaN   
Aishwarya reddy                 NaN                                    3.0   

                    Guardians of the Galaxy  The Lego Movie  The Hobbit  \
Name                                                                      
Aarti Jaiswal                           NaN             NaN         4.0   
Aditya Dharmasagar                      NaN             4.0         5.0   
Adrian Cavallaris                       3.0             NaN         4.0   
Ahmed Muheebuddin                       4.0             NaN         3.0   
Aishwarya reddy                         4.0             3.0         4.0   

                    Transformers  Malificent  Big Hero 6  Godzilla  \
Name                                                                 
Aarti Jaiswal                3.0         NaN         5.0       NaN   
Aditya Dharmasagar           4.0         4.0         NaN       4.0   
Adrian Cavallaris            2.0         NaN         NaN       NaN   
Ahmed Muheebuddin            NaN         NaN         NaN       NaN   
Aishwarya reddy              4.0         3.0         NaN       4.0   

                    Interstellar  How to Train your Dragon 2  Gone Girl  \
Name                                                                      
Aarti Jaiswal                NaN                         NaN        4.0   
Aditya Dharmasagar           5.0                         5.0        5.0   
Adrian Cavallaris            NaN                         NaN        NaN   
Ahmed Muheebuddin            NaN                         NaN        NaN   
Aishwarya reddy              4.0                         3.0        5.0   

                    Divergent  The Fault in Our Stars  Unbroken  \
Name                                                              
Aarti Jaiswal             NaN                     NaN       NaN   
Aditya Dharmasagar        4.0                     4.0       NaN   
Adrian Cavallaris         4.0                     NaN       NaN   
Ahmed Muheebuddin         NaN                     NaN       NaN   
Aishwarya reddy           NaN                     4.0       NaN   

                    300: Rise of an Empire  
Name                                        
Aarti Jaiswal                          NaN  
Aditya Dharmasagar                     5.0  
Adrian Cavallaris                      NaN  
Ahmed Muheebuddin                      NaN  
Aishwarya reddy                        3.0

In [29]:
M

American Sniper  The Hunger Games: Mockingjay - Part 1  \
Name                                                                         
Aarti Jaiswal                   4.0                                    NaN   
Aditya Dharmasagar              5.0                                    NaN   
Adrian Cavallaris               NaN                                    4.0   
Ahmed Muheebuddin               5.0                                    NaN   
Aishwarya reddy                 NaN                                    3.0   
Akhilesh                        NaN                                    NaN   
Andrew Webb                     4.0                                    3.0   
Anirudh                         3.0                                    4.0   
Anirudh Thota                   NaN                                    3.0   
anurag                          4.0                                    NaN   
Anusha Putta                    NaN                                    NaN   
Arun Vedere                     4.0                                    3.0   
Ashutosh Sureka                 5.0                                    4.0   
Aswini kumar                    4.0                                    4.0   
azam                            4.0                                    NaN   
Benjamin Baysinger              5.0                                    4.0   
Bhuvana Posina                  5.0                                    4.0   
Bradley Brubaker                NaN                                    1.0   
Brant Mullinix                  NaN                                    4.0   
Charlie                         NaN                                    2.0   
David Huang                     5.0                                    NaN   
David Reyling                   NaN                                    1.0   
Evan Piercy                     4.0                                    NaN   
Graham Bullard                  3.0                                    2.0   
Greg Werner                     5.0                                    3.0   
Harish Kallianpur               5.0                                    3.0   
Heather Sanders                 NaN                                    3.0   
HJ SONG                         NaN                                    2.0   
J A Barnes                      NaN                                    3.0   
Jake Metzger                    NaN                                    NaN   
...                             ...                                    ...   
Andrew Peabody                  NaN                                    3.0   
Matthew Dearing                 4.0                                    4.0   
Matthew Murphy                  3.0                                    NaN   
Fred Young                      3.0                                    4.0   
Rohit Tatar                     4.0                                    NaN   
Pratik Belhekar                 NaN                                    4.0   
Chris DiBello                   3.0                                    2.0   
Rajat Hatwar                    4.0                                    3.0   
Chaitanya Kusurkar              4.0                                    NaN   
Ramya Doddi                     NaN                                    4.0   
Amy Camp                        3.0                                    NaN   
Poojitha Polusani               NaN                                    NaN   
Tushar Borole                   NaN                                    3.0   
Roopa Maddikunta                4.0                                    3.5   
Josh Garrett                    5.0                                    NaN   
Koushal Yeginwar                4.0                                    NaN   
Rajesh Jaiswal                  4.0                                    3.0   
Shreya Kulkarni                 4.0                                    5.0   
Chaitanya Varma                 4.0       

In [30]:
M.columns

Index([u'American Sniper', u'The Hunger Games: Mockingjay - Part 1',
       u'Guardians of the Galaxy', u'The Lego Movie', u'The Hobbit',
       u'Transformers', u'Malificent', u'Big Hero 6', u'Godzilla',
       u'Interstellar', u'How to Train your Dragon 2', u'Gone Girl',
       u'Divergent', u'The Fault in Our Stars', u'Unbroken',
       u'300: Rise of an Empire'],
      dtype='object')

In [31]:
def pearson(s1, s2):
    """Take two pd.Series objects and return a pearson correlation."""
    s1_c = s1 - s1.mean()
    s2_c = s2 - s2.mean()
    return np.sum(s1_c * s2_c) / np.sqrt(np.sum(s1_c ** 2) * np.sum(s2_c ** 2))

In [32]:
pearson(M['American Sniper'],M['The Fault in Our Stars'])

-0.064418287383691389

Good job,pearson formula! I would never suggest 'The Fault in Our Stars' to 'American Sniper' movie guy

In [33]:
def get_recs(movie_name, M, num):

    import numpy as np
    reviews = []
    for title in M.columns:
        if title == movie_name:
            continue
        cor = pearson(M[movie_name], M[title])
        if np.isnan(cor):
            continue
        else:
            reviews.append((title, cor))
    
    reviews.sort(key=lambda tup: tup[1], reverse=True)
    return reviews[:num]

    

## • I want to know which movie is most Similar to 'The fault in our stars'

In [34]:
recs= get_recs('The Fault in Our Stars',M,1)

In [35]:
recs

[('Malificent', 0.24772975460916977)]

## • Movies I would most likely see, based on my classmates experience

In [36]:
#the movies I've seen
my_ratings = M.ix['Roopa Maddikunta'].dropna()
#the movie I've rated the highest
my_ratings.sort(ascending=False)
#the name of the movie I've rated the highest
fav_movie = my_ratings.index[0]

C:\Users\Roopa\Anaconda2\lib\site-packages\ipykernel\__main__.py:4: FutureWarning: sort is deprecated, use sort_values(inplace=True) for INPLACE sorting


In [37]:
M

American Sniper  The Hunger Games: Mockingjay - Part 1  \
Name                                                                         
Aarti Jaiswal                   4.0                                    NaN   
Aditya Dharmasagar              5.0                                    NaN   
Adrian Cavallaris               NaN                                    4.0   
Ahmed Muheebuddin               5.0                                    NaN   
Aishwarya reddy                 NaN                                    3.0   
Akhilesh                        NaN                                    NaN   
Andrew Webb                     4.0                                    3.0   
Anirudh                         3.0                                    4.0   
Anirudh Thota                   NaN                                    3.0   
anurag                          4.0                                    NaN   
Anusha Putta                    NaN                                    NaN   
Arun Vedere                     4.0                                    3.0   
Ashutosh Sureka                 5.0                                    4.0   
Aswini kumar                    4.0                                    4.0   
azam                            4.0                                    NaN   
Benjamin Baysinger              5.0                                    4.0   
Bhuvana Posina                  5.0                                    4.0   
Bradley Brubaker                NaN                                    1.0   
Brant Mullinix                  NaN                                    4.0   
Charlie                         NaN                                    2.0   
David Huang                     5.0                                    NaN   
David Reyling                   NaN                                    1.0   
Evan Piercy                     4.0                                    NaN   
Graham Bullard                  3.0                                    2.0   
Greg Werner                     5.0                                    3.0   
Harish Kallianpur               5.0                                    3.0   
Heather Sanders                 NaN                                    3.0   
HJ SONG                         NaN                                    2.0   
J A Barnes                      NaN                                    3.0   
Jake Metzger                    NaN                                    NaN   
...                             ...                                    ...   
Andrew Peabody                  NaN                                    3.0   
Matthew Dearing                 4.0                                    4.0   
Matthew Murphy                  3.0                                    NaN   
Fred Young                      3.0                                    4.0   
Rohit Tatar                     4.0                                    NaN   
Pratik Belhekar                 NaN                                    4.0   
Chris DiBello                   3.0                                    2.0   
Rajat Hatwar                    4.0                                    3.0   
Chaitanya Kusurkar              4.0                                    NaN   
Ramya Doddi                     NaN                                    4.0   
Amy Camp                        3.0                                    NaN   
Poojitha Polusani               NaN                                    NaN   
Tushar Borole                   NaN                                    3.0   
Roopa Maddikunta                4.0                                    3.5   
Josh Garrett                    5.0                                    NaN   
Koushal Yeginwar                4.0                                    NaN   
Rajesh Jaiswal                  4.0                                    3.0   
Shreya Kulkarni                 4.0                                    5.0   
Chaitanya Varma                 4.0       

In [38]:
fav_movie

'How to Train your Dragon 2'

In [39]:
M.shape

(105, 16)

In [40]:
recs1=get_recs(fav_movie,M, 16)

In [41]:
recs1

[('The Lego Movie', 0.2795149240509161),
 ('Godzilla', 0.2184387574890066),
 ('Gone Girl', 0.20606264923169826),
 ('Transformers', 0.19773222115413244),
 ('Divergent', 0.18080267690473986),
 ('Guardians of the Galaxy', 0.14884898774650113),
 ('Interstellar', 0.14348483090337),
 ('American Sniper', 0.12790702751445263),
 ('The Hobbit', 0.11937361830584169),
 ('Big Hero 6', 0.083755318927693209),
 ('The Fault in Our Stars', 0.064261122539665358),
 ('300: Rise of an Empire', 0.062387151650105714),
 ('The Hunger Games: Mockingjay - Part 1', 0.014514853337512634),
 ('Unbroken', -0.011758851023185085),
 ('Malificent', -0.035320667082435711)]

I have rated 'Malificient' 4.0 but , based on my class reviews , it says that I might not really like the movie. 

In [42]:
my_ratings

How to Train your Dragon 2               5.0
Transformers                             5.0
Godzilla                                 4.9
Big Hero 6                               4.8
Interstellar                             4.5
Malificent                               4.0
The Lego Movie                           4.0
American Sniper                          4.0
The Hobbit                               3.9
Gone Girl                                3.5
Guardians of the Galaxy                  3.5
The Hunger Games: Mockingjay - Part 1    3.5
Divergent                                3.1
Unbroken                                 3.0
The Fault in Our Stars                   3.0
Name: Roopa Maddikunta, dtype: float64

In [43]:
my_ratings.isnull()

How to Train your Dragon 2               False
Transformers                             False
Godzilla                                 False
Big Hero 6                               False
Interstellar                             False
Malificent                               False
The Lego Movie                           False
American Sniper                          False
The Hobbit                               False
Gone Girl                                False
Guardians of the Galaxy                  False
The Hunger Games: Mockingjay - Part 1    False
Divergent                                False
Unbroken                                 False
The Fault in Our Stars                   False
Name: Roopa Maddikunta, dtype: bool

In [44]:
my_ratings_nan = M.ix['Roopa Maddikunta']

In [45]:
my_ratings_nan.isnull()

American Sniper                          False
The Hunger Games: Mockingjay - Part 1    False
Guardians of the Galaxy                  False
The Lego Movie                           False
The Hobbit                               False
Transformers                             False
Malificent                               False
Big Hero 6                               False
Godzilla                                 False
Interstellar                             False
How to Train your Dragon 2               False
Gone Girl                                False
Divergent                                False
The Fault in Our Stars                   False
Unbroken                                 False
300: Rise of an Empire                    True
Name: Roopa Maddikunta, dtype: bool

## • I haven't rated only '300: Rise of an Empire' . Let's see how much I would have rated it ( I'll take help from my classmates :P) 

In [46]:
recs2=get_recs('300: Rise of an Empire',M,1)


In [47]:
recs2

[('Divergent', 0.29880439525455205)]

In [48]:
Rise_of_an_Empire_rating=M['Divergent'].mean()

In [49]:
Rise_of_an_Empire_rating

3.272881355932203

I'll most likely rate '300: Rise of an Empire' a 3.27. Well, I think now I'll watch the movie after all. 